WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -150,7 +154,8 @@ final class ProductSelectorViewModel: ObservableObject {
          toggleAllVariationsOnSelection: Bool = true,
          onProductSelected: ((Product) -> Void)? = nil,
          onVariationSelected: ((ProductVariation, Product) -> Void)? = nil,
-         onMultipleSelectionCompleted: (([Int64]) -> Void)? = nil) {
+         onMultipleSelectionCompleted: (([Int64]) -> Void)? = nil,
+         onClearedSelection: (() -> Void)? = nil) {
```

Comment:
Nit: can we name this `onSelectionsCleared` to match the pattern of other callbacks?

Reply:
❓ AFAIK we also trigger `onVariationSelected` when selecting variations. Do we need a separate closure for when the Clear selection button is tapped on the variation list but not on the product list?

Reply:
> Nit: can we name this onSelectionsCleared to match the pattern of other callbacks?
Definitely, changed: Nit: can we name this onSelectionsCleared to match the pattern of other callbacks?

> ❓ AFAIK we also trigger onVariationSelected when selecting variations. Do we need a separate closure for when the Clear selection button is tapped on the variation list but not on the product list?

You're 100% correct, not doing so causes a bug where selecting variations > tap "clear selection" > moving back to the product selector doesn't update the IDs. Fixed on 0e8540a38e25f5c1ff6f6df699b81519f7d2b6b9 by adding a similar callback as on the product selector.